source/common/stream_info/utility.cc

```diff
@@ -34,6 +35,11 @@ const std::string ResponseFlagUtils::toShortString(const StreamInfo& stream_info
 
   static_assert(ResponseFlag::LastFlag == 0x2000, "A flag has been added. Fix this code.");
 
+  // stream_info.protocol() is only available for HTTP requests.
+  if (stream_info.protocol() && !stream_info.hasAnyResponseFlag() && !stream_info.responseCode()) {
```

Comment:
Can you add a comment explaining how these conditions imply downstream disconnect? I'm not entirely convinced that these are necessary and sufficient conditions, so any hint to the reader when digging through the source to see why they are the same as a DC would help.

Reply:
Addressed in 0e4bcd5d80a778deb79aeb554870aae58297f354. Let me know if you think something else should be added/removed in the conditions.

Reply:
@venilnoronha I'm a little late to this, but FWIW I think this is pretty cludgy and will be prone to breakage in the future. Why not just have the calling code explicitly set this flag state? For example, the HCM could make this determination and set the flag? cc @htuch

Reply:
That's a great point. I've tried to address that in #5229.